apis/v1alpha1/httproute_types.go

```diff
@@ -328,79 +328,94 @@ type HTTPRouteMatch struct {
-	//   filters is defined in this package.
-	// - Custom: These filters are defined and supported by specific vendors.
+	// Type identifies the type of filter to apply. As with other API fields,
+	// types are classified into three conformance-levels:
+	//
+	// - Core: Filter types and their corresponding configuration defined by
+	//   "Support: Core" in this package, e.g. "RequestHeaderModifier". All
+	//   implementations must support core filters.
+	//
+	// - Extended: Filter types and their corresponding configuration defined by
+	//   "Support: Extended" in this package, e.g. "RequestMirror". Implementers
+	//   are encouraged to support extended filters.
+	//
+	// - Custom: Filters that are defined and supported by specific vendors.
 	//   In the future, filters showing convergence in behavior across multiple
 	//   implementations will be considered for inclusion in extended or core
-	//   conformance rings. Filter-specific configuration for such filters
+	//   conformance levels. Filter-specific configuration for such filters
 	//   is specified using the ExtensionRef field. `Type` should be set to
-	//   "ImplementationSpecific" for custom filters.
+	//   "Custom" for custom filters.
 	//
-	// Implementers are encouraged to define custom implementation
-	// types to extend the core API with implementation-specific behavior.
+	// Implementers are encouraged to define custom implementation types to
+	// extend the core API with implementation-specific behavior.
 	//
 	// +unionDiscriminator
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=100
-	Type string `json:"type"`
+	Type HTTPRouteFilterType `json:"type,omitempty"`
 
-	// ExtensionRef is an optional, implementation-specific extension to the
-	// "filter" behavior.  The resource may be "configmap" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myroutefilters" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".
-	// ExtensionRef MUST NOT be used for core and extended filters.
+	// RequestHeaderModifier defines a schema for a filter that modifies request
+	// headers.
+	//
+	// Support: Core
+	//
 	// +optional
-	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
-
-	// Filter-specific configuration definitions for core and extended filters
+	RequestHeaderModifier *HTTPRequestHeaderFilter `json:"requestHeaderModifier,omitempty"`
 
+	// RequestMirror defines a schema for a filter that mirrors requests.
+	//
+	// Support: Extended
+	//
 	// +optional
-	RequestHeader *HTTPRequestHeaderFilter `json:"requestHeader,omitempty"`
+	RequestMirror *HTTPRequestMirrorFilter `json:"requestMirror,omitempty"`
 
+	// ExtensionRef is an optional, implementation-specific extension to the
+	// "filter" behavior.  The resource may be "configmap" or an
```

Comment:
Recently, we have intentionally removed references to ConfigMaps in extension points.
Our recent stance has been that there is nothing stopping users from using ConfigMap but we don't want to encourage this pattern.

Reply:
This was already present in the code before this PR so this could be considered as a small scope creep for this PR.